Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -572,10 +572,22 @@ A convenient way to dynamically test deep linking is to use Frida or frida-trace
+Android considers [different types of links](https://developer.android.com/training/app-links):
+
+- **Deep links**: unverified URIs of any scheme that take users directly to specific content in an app.
+- **Web Links**: unverified deep links that use the HTTP and HTTPS schemes.
+- **Android App Links** (Android 6.0 (API level 23) and higher): verified web links that use the HTTP and HTTPS schemes and contain the `autoVerify` attribute.
+
+#### Deep Links and Custom URL Schemes
+
+An app can [set up deep links](https://developer.android.com/training/app-links/deep-linking) by adding _intent filters_ on the Android Manifest and extracting data from incoming intents to navigate users to the correct activity. It can even use any custom scheme prefix such as `myapp`, which will result in the URI prefix "myapp://". These kinds of deep links are also referred to as "Custom URL Schemes" and are typically used as a form of inter-app communication where an app can define certain actions (including the corresponding parameters) that can be triggered by other apps.
 
 This method of defining deep links via intent filters has an important issue: any other apps installed on a user's device can declare and try to handle the same intent (typically a custom URL scheme). This is known as **deep link collision** where any arbitrary application can declare control over the exact same URL custom scheme belonging to another application. In recent versions of Android this results in a so-called _disambiguation dialog_ being shown to the user and asking them to select the application that should handle the link. The user could make the mistake of choosing a malicious application instead of the legitimate one.
 
+![OWASP_MSTG](https://developer.android.com/training/app-links/images/app-disambiguation_2x.png)
+
+> Something similar happens with Android App Links. Only one app at a time can be associated with a particular domain. If another app is already verified for the domain, the user must first disassociate that other app with the domain before they can associate your app with the domain. Apps are able to [ask the user to assotiate an app with a certain domain](https://developer.android.com/training/app-links/verify-site-associations#request-user-associate-app-with-domain). However, it is not a risk since these links are verified and will only work with the declaring legitimate apps.
```

Comment:
```suggestion
Something similar happens with Android App Links. Only one app at a time can be associated with a particular domain. If another app is already verified for the domain, the user must first disassociate that other app with the domain before they can associate your app with the domain. Apps are able to [ask the user to assotiate an app with a certain domain](https://developer.android.com/training/app-links/verify-site-associations#request-user-associate-app-with-domain). However, it is not a risk since these links are verified and will only work with the declaring legitimate apps.
```